File: app/src/main/java/in/woloo/www/interestedtopic/adapter/InterestedTopicAdapter.java
package in.woloo.www.interestedtopic.adapter;

import in.woloo.www.R;
import in.woloo.www.trendingblog.model.CategoriesResponse;
import in.woloo.www.utils.ImageUtil;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.CheckBox;
import android.widget.ImageView;
import android.widget.RelativeLayout;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.bumptech.glide.Glide;

import org.jetbrains.annotations.NotNull;

import java.util.ArrayList;
import java.util.List;

public class InterestedTopicAdapter extends RecyclerView.Adapter<InterestedTopicAdapter.ViewHolder> {

    private final LayoutInflater inflater;
    private final List<CategoriesResponse.Category> categories;
    private static OnItemCheckListener itemCheckListener;
    Context context;

    public InterestedTopicAdapter(Context context, List<CategoriesResponse.Category> categories) {
        this.inflater = LayoutInflater.from(context);
        this.categories = categories;
        this.context = context;
        if (context instanceof OnItemCheckListener) {
            itemCheckListener = (OnItemCheckListener) context;
        }
    }

    @NonNull
    @NotNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull @NotNull ViewGroup parent, int viewType) {
        View view = inflater.inflate(R.layout.recycler_gridlayout_design, parent, false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull @NotNull InterestedTopicAdapter.ViewHolder holder, int position) {
        CategoriesResponse.Category category = categories.get(position);
//        ImageUtil.loadImage(holder.topicImage.getContext(), holder.topicImage, category.getCategoryIconUrl());
        Glide.with(context)
                .load(category.getCategoryIconUrl())
                .error(context.getDrawable(R.drawable.ic__01_hormones))
                .into(holder.topicImage);
        holder.textTitle.setText(category.getCategoryName());
        holder.parentView.setOnClickListener(view -> {
            if (itemCheckListener != null) {
                itemCheckListener.onItemClick(position);
            }
            category.setSelected(!category.isSelected());
            if (category.isSelected()) {
                holder.imageRelative.setBackgroundResource(R.drawable.circular_background_change);
            } else {
                holder.imageRelative.setBackgroundResource(R.drawable.circular_background);
            }
        });
        if (category.isSelected()) {
            holder.imageRelative.setBackgroundResource(R.drawable.circular_background_change);
        } else {
            holder.imageRelative.setBackgroundResource(R.drawable.circular_background);
        }
    }

    @Override
    public int getItemCount() {
        return categories.size();
    }

    public static class ViewHolder extends RecyclerView.ViewHolder {
        ImageView topicImage;
        TextView textTitle;
        RelativeLayout imageRelative;
        View parentView;

        public ViewHolder(@NonNull @NotNull View itemView) {
            super(itemView);
            parentView = itemView;
            textTitle = itemView.findViewById(R.id.itemNameText);
            topicImage = itemView.findViewById(R.id.checkboxItem);
            imageRelative = itemView.findViewById(R.id.gridDesignRel1);
        }
    }

    public interface OnItemCheckListener {
        void onItemClick(int position);
    }
}

File: app/src/main/java/in/woloo/www/application_kotlin/adapters/login/InterestedTopicAdapter.kt
package `in`.woloo.www.application_kotlin.adapters.login

import android.content.Context
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.LinearLayout
import android.widget.RelativeLayout
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide
import `in`.woloo.www.R
import `in`.woloo.www.more.trendingblog.model.CategoriesResponse

class InterestedTopicAdapter(
    var context: Context,
    private val categories: List<CategoriesResponse.Category>
) :
    RecyclerView.Adapter<InterestedTopicAdapter.ViewHolder>() {
    private val inflater: LayoutInflater

    init {
        inflater = LayoutInflater.from(context)
        if (context is OnItemCheckListener) {
            itemCheckListener = context as OnItemCheckListener
        }
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = inflater.inflate(R.layout.recycler_gridlayout_design, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val category = categories[position]
        //        ImageUtil.loadImage(holder.topicImage.getContext(), holder.topicImage, category.getCategoryIconUrl());
        Log.d("url is" , category.categoryIconUrl!!);
        Glide.with(context)
            .load(category.categoryIconUrl)
            .error(context.getDrawable(R.drawable.ic__01_hormones))
            .into(holder.topicImage)
        holder.textTitle.text = category.categoryName
        holder.parentView.setOnClickListener { view: View? ->
            if (itemCheckListener != null) {
                itemCheckListener!!.onItemClick(position)
            }
            category.isSelected = !category.isSelected
            if (category.isSelected) {
                holder.cardRelative.setBackgroundResource(R.drawable.yellow_border_white_rect)
            } else {
                holder.cardRelative.setBackgroundResource(R.drawable.new_button_background)
            }
        }
        if (category.isSelected) {
            holder.cardRelative.setBackgroundResource(R.drawable.yellow_border_white_rect)
        } else {
            holder.cardRelative.setBackgroundResource(R.drawable.new_button_background)
        }
    }

    override fun getItemCount(): Int {
        return categories.size
    }

    class ViewHolder(var parentView: View) : RecyclerView.ViewHolder(
        parentView
    ) {
        var topicImage: ImageView
        var textTitle: TextView
        var cardRelative: LinearLayout

        init {
            textTitle = itemView.findViewById(R.id.itemNameText)
            topicImage = itemView.findViewById(R.id.checkboxItem)
            cardRelative = itemView.findViewById(R.id.relative_card)
        }
    }

    interface OnItemCheckListener {
        fun onItemClick(position: Int)
    }

    companion object {
        private var itemCheckListener: OnItemCheckListener? = null
    }
}
